etc/kayobe/ansible/get-cloud-facts.yml

```
@@ -0,0 +1,124 @@
+---
+- name: Get facts for control host
+  hosts: localhost
+  gather_facts: true
+  tasks:
+    - name: Check if ngs enabled
+      ansible.builtin.set_fact:
+        ngs_enabled: >-
+          {{ 'genericswitch' in kolla_neutron_ml2_mechanism_drivers | default(false) }}
+      when: kolla_neutron_ml2_mechanism_drivers
+
+- name: Get facts for seed node
+  hosts: seed
+  gather_facts: true
+  tasks:
+    - name: Is there a dedicated seed node?
+      ansible.builtin.set_fact:
+        seed_node_is_vm: "{{ ansible_facts.virtualization_role == 'guest' }}"
+
+- name: Get facts
+  hosts: all
+  gather_facts: true
+  tasks:
+    - name: Pinging stackhpc.com to check internet connectivity
+      ansible.builtin.command: "ping stackhpc.com -c 3"
+      register: successful_ping
+      timeout: 5
+      ignore_errors: true
+
+    - name: Set internet connectivity fact
+      ansible.builtin.set_fact:
+        internet_connectivity: "{{ not successful_ping.failed }}"
+    - name: Get OS distribution
+      ansible.builtin.set_fact:
+        distribution: "{{ ansible_facts.distribution }}"
+
+
+    - name: Get OS distribution release
+      ansible.builtin.set_fact:
+        distribution_release: "{{ ansible_facts.distribution_release }}"
+
+
+    - name: Get kernel version
+      ansible.builtin.command: "uname -r"
+      register: kernel_version
+
+    - name: Check if docker is installed
+      ansible.builtin.command: "which docker"
+      register: docker_check
+      failed_when: docker_check.rc not in [0]
+      ignore_errors: true
+
+    - name: Get running contianers
+      ansible.builtin.command: "docker ps --format {% raw %}'{{.Names}}'{% endraw %}"
+      become: true
+      register: containers_list
+      when: not docker_check.failed
+
+    - name: Check if hugepages enabled
+      ansible.builtin.set_fact:
+        hugepages_enabled: "{{ 'hugepages' in ansible_facts.cmdline }}"
+
+- name: Gather Cloud Facts
```

The name of this play is now a bit similar to the previous "gather facts" plays. I'd rename it to be a bit more clear 
```suggestion
- name: Write cloud facts file
```